pkg/reconciler/namespace/namespace.go

```diff
@@ -187,6 +187,11 @@ func (c *Controller) ensureScheduled(ctx context.Context, ns *corev1.Namespace)
 	patchType, patchBytes := clusterLabelPatchBytes(newPClusterName)
 	_, err = c.kubeClient.Cluster(ns.ClusterName).CoreV1().Namespaces().
 		Patch(ctx, ns.Name, patchType, patchBytes, metav1.PatchOptions{})
+	if err != nil {
+		// Update the label to enable the caller to detect a scheduling change.
+		ns.Labels[ClusterLabel] = newPClusterName
```

Comment:
don't we get the namespace back in the Patch call? Would rather return that as it is guaranteed to be the real truth.

Reply:
I'm also curious about this scenario - if the patch fails, we're forcing the ClusterLabel to be what was calculated in this func, but failed to patch?

Reply:
Done.

Reply:
Do you mean `err == nil`?

Reply:
Yes! `<facepalm>`
Done.